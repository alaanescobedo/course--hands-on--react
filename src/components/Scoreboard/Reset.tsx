import styled from "@emotion/styled";
import { FC } from "react";
import { useMouse } from '../hooks/useMouse';

export interface ResetProps {
  /**
   * Reset action handler
   */
  onReset: () => void;
}

export const Reset: FC<ResetProps> = ({ onReset }) => {
  const [mouseDown, onMouseDown, onMouseUp] = useMouse();

  return (
    <Button
      onClick={onReset}
      onMouseDown={onMouseDown}
      onMouseUp={onMouseUp}
      onMouseLeave={onMouseUp}
    >
      {mouseDown ? "ðŸ˜®" : "ðŸ™‚"}
    </Button>
  );
};

const Button = styled.button`
  font-size: 1.5vw;
  cursor: pointer;
  font-size: 700;
  border-width: 0.15vw;
  border-style: solid;
  background-color: #d1d1d1;
  border-color: #fff #9e9e9e9e #9e9e9e9e #fff;
`;
