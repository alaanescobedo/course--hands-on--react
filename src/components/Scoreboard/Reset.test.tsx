import { Reset } from "./Reset";

import { render, screen, fireEvent } from "@testing-library/react";

describe("Reset Component", () => {
  const onReset = jest.fn();

  beforeEach(() => {
    render(<Reset onReset={onReset} />);
  });

  test("should render elements with default state", () => {
    expect(screen.getByText("ðŸ™‚")).toBeInTheDocument();
  });
  test("onReset handler should be called ", () => {
    fireEvent.click(screen.getByText("ðŸ™‚"));
    expect(onReset).toHaveBeenCalledTimes(1);
  });
  test("should change state when onMouseDown and onMouseUp events happened", () => {
    fireEvent.mouseDown(screen.getByText("ðŸ™‚"));
    expect(screen.getByText("ðŸ˜®")).toBeInTheDocument();

    fireEvent.mouseUp(screen.getByText("ðŸ˜®"));
    expect(screen.getByText("ðŸ™‚")).toBeInTheDocument();
  });
  test("should change state when onMouseDown and onMousesLeave events happened", () => {
    fireEvent.mouseDown(screen.getByText("ðŸ™‚"));
    expect(screen.getByText("ðŸ˜®")).toBeInTheDocument();

    fireEvent.mouseLeave(screen.getByText("ðŸ˜®"));
    expect(screen.getByText("ðŸ™‚")).toBeInTheDocument();
  });
});
