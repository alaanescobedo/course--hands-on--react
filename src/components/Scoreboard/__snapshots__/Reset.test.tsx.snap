// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Component Reset renders correctly 1`] = `
<DocumentFragment>
  <button
    class="css-15ja8qh"
  >
    ðŸ™‚
  </button>
</DocumentFragment>
`;
